qcodes/instrument/visa.py

```diff
@@ -139,12 +141,22 @@ def device_clear(self) -> None:
         if self.visabackend == 'sim':
             return
 
-        if isinstance(self.visa_handle, pyvisa.resources.SerialInstrument):
-            self.visa_handle.flush(
-                vi_const.BufferOperation.discard_read_buffer_no_io | vi_const.BufferOperation.discard_write_buffer
+        assert isinstance(self.visa_handle, pyvisa.resources.SerialInstrument)
+        if pyvisa_is_1_11_or_higher:
+            flush_operation = (
+                    vi_const.BufferOperation.discard_read_buffer
+                    | vi_const.BufferOperation.discard_write_buffer
             )
         else:
-            self.visa_handle.clear()
+
+            # This can be dropped once we drop support for qcodes <0.19
+            # and therefore pyvisa 1.10
+            flush_operation = cast(
+                Any,
+                vi_const.VI_READ_BUF | vi_const.VI_WRITE_BUF_DISCARD
+            )
+        self.visa_handle.flush(flush_operation)
+        self.visa_handle.clear()
```

Comment:
Is it okay to flush and clear or do we still need the logic to only flush if `isinstance(self.visa_handle, pyvisa.resources.SerialInstrument)` ?

Reply:
I think the flush operation only appies to serial instruments

Reply:
indeed, flush needs to be kept inside `if isinstance(self.visa_handle, pyvisa.resources.SerialInstrument):` block

Reply:
thanks for confirming